pkg/storage/mongo/mongo.go

```diff
@@ -23,9 +25,10 @@ func NewStorage(url string) *ModuleStore {
 
 // Connect conntect the the newly created mongo backend.
 func (m *ModuleStore) Connect() error {
-	s, err := mgo.Dial(m.url)
+	timeout := env.MongoConnectionTimeoutSecWithDefault(1)
+	s, err := mgo.DialWithTimeout(m.url, timeout)
 	if err != nil {
-		return err
+		return fmt.Errorf("failed to connect with %s, error: %v", m.url, err)
```

Comment:
we have an issue for more granular timeouts i think this is one of the cases when it can be used. 
Timeout for a request should not be the same as a timeout for dialing a connection.

Reply:
Do you want this to be solved in this PR, or we could raise a separate PR which handles the `storage/mongo.go` database name, this granular dial connection env vars, and other required vars.

Reply:
nono, just pointing out so we don't forget